sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -1490,5 +1496,366 @@ def pipeline_with_input(boolean: bool = False):
+            """Dummy component that outputs metrics with a random accuracy."""
+            import random
+            result = random.randint(0, 100)
+            metrics.log_metric('accuracy', result)
+
+        @dsl.pipeline(name='Test pipeline')
+        def my_pipeline(sample_input1: bool,
+                        sample_input2: str,
+                        sample_input3: Input[Model],
+                        sample_input4: float = 3.14,
+                        sample_input5: list = [1, 2, 3],
+                        sample_input6: dict = {
+                            'one': 1,
+                            'two': 2,
+                            'three': 3
+                        },
+                        sample_input7: int = 5) -> Model:
+            """This is a definition of this pipeline."""
+
+            task = output_model()
+            return task.output
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            pipeline_spec_path = os.path.join(tmpdir, 'output.yaml')
+            compiler.Compiler().compile(
+                pipeline_func=my_pipeline, package_path=pipeline_spec_path)
+
+            with open(pipeline_spec_path, 'r+') as f:
+                yaml_content = f.read()
+
+        predicted_comment = textwrap.dedent("""\
+                # PIPELINE DEFINITION
+                # Name: test-pipeline
+                # Description: This is a definition of this pipeline.
+                # Inputs:
+                #    sample_input1: bool
+                #    sample_input2: str
+                #    sample_input3: system.Model
+                #    sample_input4: float [Default: 3.14]
+                #    sample_input5: list [Default: [1.0, 2.0, 3.0]]
+                #    sample_input6: dict [Default: {'one': 1.0, 'three': 3.0, 'two': 2.0}]
+                #    sample_input7: int [Default: 5.0]
+                # Outputs:
+                #    Output: system.Model
+                """)
+
+        # test predicted comment matches actual comment
+        self.assertIn(predicted_comment, yaml_content)
+
+    def test_comments_on_compiled_components(self):
+
+        @dsl.component
+        def my_component(string: str, model: bool) -> str:
+            return string
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            pipeline_spec_path = os.path.join(tmpdir, 'output.yaml')
+            compiler.Compiler().compile(
+                pipeline_func=my_component, package_path=pipeline_spec_path)
+
+            with open(pipeline_spec_path, 'r+') as f:
+                yaml_content = f.read()
+
+        predicted_comment = textwrap.dedent("""\
+                # PIPELINE DEFINITION
+                # Name: my-component
+                # Inputs:
+                #    model: bool
+                #    string: str
+                """)
+
+        # test comments work on compiled components
+        self.assertIn(predicted_comment, yaml_content)
+
+        @dsl.container_component
```

Comment:
Consider adding a description test for the container component also.

Reply:
Added a description to the current container component test case, that way it tests all properties. If you have a strong opinion in making it independent i could also do that too

Reply:
SGTM